liboptv/tests/check_track.c

```diff
@@ -0,0 +1,558 @@
+
+
+}
+END_TEST
+
+
+START_TEST(test_sort_candidates_by_freq)
+{
+    foundpix src[] = {  {1,0,{1,0}},
+                        {2,0,{1,1}}
+                     };
+    foundpix *dest;
+    int num_cams = 2;
+    
+    /* sortwhatfound freaks out if array is not reset before */
+    dest = (foundpix *) calloc (num_cams*MAX_CANDS, sizeof (foundpix));
+    reset_foundpix_array(dest, num_cams*MAX_CANDS, num_cams);
+    copy_foundpix_array(dest, src, 2, 2);
+
+    
+    int counter;
+
+    /* test simple sort of a small foundpix array */
+    sort_candidates_by_freq(dest, &counter, num_cams);
+    
+    ck_assert_msg( dest[0].ftnr == 2 ,
+                  "Was expecting dest[0].ftnr == 2 but found %d \n", dest[0].ftnr);
+    ck_assert_msg( dest[0].freq == 2 ,
+                  "Was expecting dest[0].freq == 2 but found %d \n", dest[0].freq);
+    ck_assert_msg( dest[1].freq == 0 ,
+                  "Was expecting dest[1].freq == 0 but found %d \n", dest[1].freq);
+
+}
+END_TEST
+
+START_TEST(test_trackcorr_c_loop)
+{
+    tracking_run *ret;
+    int step, display=0;
+    Calibration *calib[3];
+    control_par *cpar;
+
+    chdir("testing_fodder/track");
+    
+    printf("----------------------------\n");
+    printf("Test tracking multiple files 2 cameras, 1 particle \n");
+    
+    cpar = read_control_par("parameters/ptv.par");
+    read_all_calibration(calib, cpar->num_cams);
+    ret = trackcorr_c_init(calib);
+
+
+    trackcorr_c_loop (ret, ret->seq_par->first, display, calib);
+    
+    for (step = ret->seq_par->first+1; step < ret->seq_par->last; step++)
+    {
+        trackcorr_c_loop (ret, step, display, calib);
+    }
+    trackcorr_c_finish(ret, ret->seq_par->last, display);
+    
+    int range = ret->seq_par->last - ret->seq_par->first;
+    double npart, nlinks;
+    
+    /* average of all steps */
+    npart = (double)ret->npart / range;
+    nlinks = (double)ret->nlinks / range;
+    
+    ck_assert_msg(fabs(npart - 208.0/210.0)<EPS,
+                  "Was expecting npart == 208/210 but found %f \n", npart);
+    ck_assert_msg(fabs(nlinks - 198.0/210.0)<EPS,
+                  "Was expecting nlinks == 198/210 but found %f \n", nlinks);
+    
+}
+END_TEST
+
+START_TEST(test_cavity)
+{
+    tracking_run *ret;
+    int display=0;
+    Calibration *calib[4];
+    control_par *cpar;
+    
+    
+    printf("----------------------------\n");
+    printf("Test cavity case \n");
+    
+    chdir("testing_fodder/test_cavity");
+    
+    
+    cpar = read_control_par("parameters/ptv.par");
+    read_all_calibration(calib, cpar->num_cams);
+    printf("In test_cavity num cams = %d\n",cpar->num_cams);
+    ret = trackcorr_c_init(calib);
+    
+    trackcorr_c_loop (ret, 10002, display, calib);
+    //trackcorr_c_finish(ret, 10002, display);
+    
+    ck_assert_msg(ret->npart == 672,
+                  "Was expecting npart == 672 but found %f \n", ret->npart);
+    ck_assert_msg(ret->nlinks == 99,
```

Comment:
I try to run now the old PyPTV to see if it gives the same number. as far as I remember this is it for the old 3D-PTV (Tcl/Tk) version that I try to compile as well.